### PR TITLE
TINKERPOP-2089 Added DSL support to gremlin-javascript

### PR DESCRIPTION
**Pull request by spmallette**
https://issues.apache.org/jira/browse/TINKERPOP-2089

Pretty straightforward, but it feels a bit hacked together with my limited javascript skills. I imagine cleanup from the experts will be needed before this is all said and done. Submitting this as a draft for now - will complete docs and such once the code is "right".

### CODE REVIEW

File: gremlin-javascript/glv/GraphTraversalSource.template [hunk 1]
```diff
@@ -51,7 +55,8 @@ class GraphTraversalSource {
   withRemote(remoteConnection) {
     const traversalStrategy = new TraversalStrategies(this.traversalStrategies);
     traversalStrategy.addStrategy(new remote.RemoteStrategy(remoteConnection));
-    return new GraphTraversalSource(this.graph, traversalStrategy, new Bytecode(this.bytecode));
+    var object = Object.create(this.constructor.prototype);
```
Comment (jorgebay) [inline gremlin-javascript/glv/GraphTraversalSource.template]: Looks like a leftover.

File: gremlin-javascript/src/main/javascript/gremlin-javascript/lib/process/anonymous-traversal.js [hunk 2]
```diff
@@ -31,13 +31,17 @@ const Graph = require('../structure/graph').Graph;
...
+  }
+
   /**
    * Constructs an {@code AnonymousTraversalSource} which will then be configured to spawn a
    * {@link GraphTraversalSource}.
    * @returns {AnonymousTraversalSource}.
    */
-  static traversal() {
-    return new AnonymousTraversalSource();
+  static traversal(traversalSourceClass) {
```
Comment (jorgebay) [inline gremlin-javascript/src/main/javascript/gremlin-javascript/lib/process/anonymous-traversal.js]: We should document in jsdoc: `@param {GraphTraversalSource} [traversalSourceClass] Optional GraphTraversalSource constructor.`

File: gremlin-javascript/glv/GraphTraversalSource.template (hunk 1 above)
Comment (spmallette) [reply gremlin-javascript/glv/GraphTraversalSource.template]: yeah - that should be deleted.

**Comment by spmallette** [comment]
@jorgebay  I'll wait for you to push the fixes you intended to add. I just realized though that I didn't handle testing anonymous traversals. I still need to go back in and do that.

> but after seeing how clean a custom DSL would look like (SocialTraversal and SocialTraversalSource), I think it makes sense to match the functionality in the other GLVs

great!

**Comment by jorgebay** [comment]
I've pushed a commit to the dev branch, feel free to override.

**Pull request merged**